src/main/java/com/xpanxion/java/springboot/da1/demo/service/student2/HistoryService2.java

```diff
@@ -27,6 +32,27 @@
     @Autowired
     private CheckoutRepository2 checkoutRepository2;
 
+    public List<WorkoutHistory2> getHistory(int memberId) {
+
+        workoutHistory.clear();
+
+        List<Checkin2> checkinHistory = checkinRepository2.findBymember2Id(memberId);
+        List<Checkout2> checkoutHistory = checkoutRepository2.findBymember2Id(memberId);
+
+        for (Checkin2 n : checkinHistory) {
+            workoutHistory.add(new WorkoutHistory2(memberId, n.getCheckin(), "CHECK_IN"));
+        }
+        for (Checkout2 n : checkoutHistory) {
+            workoutHistory.add(new WorkoutHistory2(memberId, n.getCheckout(), "CHECK_OUT"));
+        }
+
+        if (workoutHistory.isEmpty()) { // Feels wasteful to put it after for loops but an easy temporary solution
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "member id not found");
```

Comment:
I'm not sure if this makes sense.  So... if a member has joined the join, but hasn't worked out yet, will this endpoint return a `404 member id not found`?